sbot/motor_board.py

```diff
@@ -0,0 +1,294 @@
+"""The motor board module provides an interface to the motor board firmware over serial."""
+from __future__ import annotations
+
+import atexit
+import logging
+from enum import IntEnum
+from types import MappingProxyType
+from typing import NamedTuple
+
+from serial.tools.list_ports import comports
+
+from .exceptions import BoardDisconnectionError, IncorrectBoardError
+from .logging import log_to_debug
+from .serial_wrapper import SerialWrapper
+from .utils import (
+    Board, BoardIdentity, float_bounds_check,
+    get_USB_identity, map_to_float, map_to_int,
+)
+
+logger = logging.getLogger(__name__)
+BAUDRATE = 115200
+
+
+class MotorPower(IntEnum):
+    """Special values for motor power."""
+    BRAKE = 0
+    COAST = -1024  # A value outside the allowable range
+
+
+class MotorStatus(NamedTuple):
+    """A tuple representing the status of the motor board."""
+    output_faults: tuple[bool, ...]
+    input_voltage: float
+
+    @classmethod
+    def from_status_response(cls, response: str) -> MotorStatus:
+        """
+        Create a MotorStatus object from the response to a status command.
+
+        :param response: The response from a *STATUS? command.
+        :return: A MotorStatus object.
+        """
+        data = response.split(':')
+        output_faults = tuple((port == '1') for port in data[0].split(','))
+        input_voltage = float(data[1]) / 1000
+        return cls(output_faults, input_voltage)
```

Comment:
```suggestion
        output_fault_list, input_voltage_raw = response.split(':', 2)
        output_faults = tuple((port == '1') for port in output_fault_list.split(','))
        input_voltage = float(input_voltage_raw) / 1000
        return cls(output_faults, input_voltage)
```
Generally, it's best practise to use the `maxsplit` arg when parsing like this.

There are several potential `TypeError`s in this function, perhaps if we not handling them, we should declare that it could be thrown in the docstring?

Reply:
The issue here though is that having extra elements in the status is not an error. 

Allowing the status to contain extra fields allows us to get extra information out of experimental firmwares without needing to modify sbot.

Reply:
We wouldn't be able to get to the extra data, as we'd only be parsing out the values we expect (assuming their formats are unchanged). With that said, I think discarding the remaining parts of the message feels perfectly fine.